kliff/models/kim.py

```diff
@@ -576,21 +540,58 @@ def set_one_opt_param(self, name: str, settings: List[List[Any]]):
 
         Args:
             name: name of a fitting parameter
-            settings: initial value, flag to fix a parameter, lower and upper bounds of a
+            settings: List initial value, flag to fix a parameter, lower and upper bounds of a
                 parameter.
 
         Example:
-            name = 'param_A'
+            name = 'A'
             settings = [['default', 0, 20], [2.0, 'fix'], [2.2, 'inf', 3.3]]
             instance.set_one(name, settings)
         """
-        self.opt_params.set_one(name, settings)
+        # Keeping this API intact for now, no improvement comes to mind
+        # self.opt_params.set_one(name, settings)
+        param = self.model_params[name]
+        # check the val kind
+        supplied_value = settings[0][0]
+        if supplied_value == "default":
+            supplied_value = param.get_numpy_array_model_space()
+        elif isinstance(supplied_value, (int, float)):
+            supplied_value = np.array([supplied_value])
+        elif isinstance(supplied_value, Parameter):
+            supplied_value = supplied_value.get_numpy_array_model_space()
+        else:
+            raise ValueError("Settings array is not properly formatted")
+        # When model is operating with transformed parameters
+        # input is expected in transformed space
+        param.copy_from_param_space(supplied_value)
+
+        # replace "inf" with np.inf
+        if len(settings[0]) > 1:
+            setting_ = [val if val != "inf" else np.inf for val in settings[0]]
+            if len(setting_) == 3:
+                bounds = [setting_[1], setting_[2]]
+            elif setting_[1] == "fix":
```

Comment:
`fix` means a value will not be optimized. We need to modify `opt_mask` to reflect this. 

For example, `[2.0, "fix"]`, means we want to set the model parameter to 2.0, (the default might be different), but we want it to stay at 2.0 and not optimize. 

Many lines of the code here are similar to that in `set_one_opt_param` of `model.py`. To simplify it, can we combine and move the bulk to `model.py`? We can then use `super()` to call it here and add additional code specific to kim models.

Reply:
I think we can revisit it in the end after having more complete picture of model / KIMmodel division. Problem is Model class is very much inherited by KIMModel only with limited usage in LennardJones. Thus I am not sure how exactly to structure Model and KIMModel. To keep it general, parameters are marked optimizable (by modifying their opt_mask) in `Model.set_opt_param` function, as that interfaces with the Parameter class and have to be more generic. 

Updating model parameters is model dependent hence, `Model` has just basic support of copying values. Whereas KIMModel has specific, KIMPY based update overrides. I would recommend that we create an Issue out of this conversation thread and revisit it in the end to see how to organize it further.

Reply:
Similar issue is with `set_param_mutable` function, where it not clear that what is the exact vision for `Model` class, as only KIMModel seem to use it seriously.

Reply:
Yup. Fixed. Now most of duplicated code lives in Model.py, and KIMModel calls the `super()` method. KIMModel then copies the values to KIMPY instance.